---
title: "Khudji tooth decontamination"
author: "Zandra Fagernäs"
date: "16.10.2024"
---

This R Notebook contains code for the Khudji tooth decontamination portion of "Cleaning the Dead: Optimized decontamination enhances palaeoproteomic analyses of a Pleistocene hominin tooth from Khudji, Tajikistan" by Fagernäs et al.

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(MetBrewer)
library(readr)
library(janitor)
library(ggpubr)
library(Hmisc)
library(lme4)
library(lmerTest)
library(MASS)
library(Peptides)
library(tidyverse)
```

The first part of the analysis is based on MaxQuant data output, which has been deposited to the ProteomeXchange Consortium via the PRIDE partner repository with the dataset identifier PXD050370. To run the code, you need to download the files from PRIDE, unzip them, and move the folders into the same folder as you have saved this script. There are two different folders, one for the bleached sample and one for the original sample. Note that the figures may have been manually edited for improved visuals after exporting.

Please note that during the MaxQuant run, the bleached sample was labelled as "Blank" due to the raw file name containing the word "blank" by accident. It is, however, not the blank. See the metadata file if you are unsure.

## Protein count

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.width=4, fig.height=4}
# Load text file with protein groups and clean 
original <- read_delim("khudji_original/proteinGroups.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!str_detect(protein_names, "Trypsin")) %>%
  filter(!str_detect(protein_i_ds, "CON__P00761")) %>%
  dplyr::select(protein_i_ds, protein_names, razor_unique_peptides_p0441_77min) %>%
  filter(razor_unique_peptides_p0441_77min > 1)
  
bleach <- read_delim("khudji_bleach/proteinGroups.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+"))  %>%
  filter(!str_detect(protein_names, "Trypsin")) %>%
  filter(!str_detect(protein_i_ds, "CON__P00761")) %>%
  dplyr::select(protein_i_ds, protein_names, razor_unique_peptides_blank) %>%
  filter(razor_unique_peptides_blank > 1)

# Combine files
protein <- full_join(original, bleach)

# Assign origin of protein
protein <- protein %>%
  mutate(skin = str_detect(protein_names, 
       "Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin")) %>%
  mutate(bone = str_detect(protein_names, "Collagen|EMI")) %>%
  mutate(other = str_detect(protein_names, "Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger")) %>%
  mutate(tissue = if_else(skin == TRUE, "skin", 
                          if_else(bone == TRUE, "bone", "other"))) %>%
  dplyr::select(-skin, -bone, -other)

# Long form
protein <- protein %>%
  pivot_longer(names_to = "sample", values_to = "peptides", 3:4) %>%
  filter(peptides > 0)

# Clean up sample names
protein <- protein %>%
    mutate(sample = gsub('razor_unique_peptides_p0441_77min', 'Original', sample)) %>%
    mutate(sample = gsub('razor_unique_peptides_blank', 'Bleach', sample))

# Count proteins 
protein_summary <- protein %>%
  group_by(sample, tissue) %>%
  dplyr::summarize(proteins = sum(peptides > 0),
            peptides = sum(peptides))

# Long format
protein_summary <- protein_summary %>%
  pivot_longer(names_to = "measure", values_to = "count", 3:4)

# Plot (creates Figure 6B)
ggplot(protein_summary, aes(x=sample, y=count, fill=tissue)) +
  geom_bar(position="fill", stat = "identity") +
  facet_wrap(~measure, scales = "free_y") +
  scale_fill_manual(values=met.brewer("Cassatt2", 3)) +
  theme_bw()
```

## PSM count

```{r}
# Load dataset
original <- read_delim("khudji_original/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!str_detect(protein_names, "Trypsin")) %>%
  filter(!str_detect(proteins, "CON__P00761")) %>%
  filter(experiment == "P0441_77min") %>%
  filter(intensity > 0) %>%
  dplyr::select(proteins, protein_names, experiment, intensity)
  
bleach <- read_delim("khudji_bleach/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+"))  %>%
  filter(!str_detect(protein_names, "Trypsin")) %>%
  filter(!str_detect(proteins, "CON__P00761")) %>%
  filter(experiment == "Blank") %>%
  filter(intensity > 0) %>%
  dplyr::select(proteins, protein_names, experiment, intensity)

# Combine files
psm <- full_join(original, bleach)

# Assign origin of protein
psm <- psm %>%
  mutate(skin = str_detect(protein_names, 
       "Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin")) %>%
  mutate(bone = str_detect(protein_names, "Collagen|EMI")) %>%
  mutate(other = str_detect(protein_names, "Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger")) %>%
  mutate(tissue = if_else(skin == TRUE, "skin", 
                          if_else(bone == TRUE, "bone", "other"))) %>%
  dplyr::select(-skin, -bone, -other)

# PSM count
psm_count <- psm %>%
  group_by(experiment, tissue) %>%
  dplyr::summarize(psm_count = n())
```

## Peptide length 

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Load data
original <- read_delim("khudji_original/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!experiment == "P0442_77min_Blank") %>%
  dplyr::select(length, experiment, intensity) %>%
  mutate(experiment = gsub('P0441_77min', 'Original', experiment))
original_bone <- read_delim("khudji_original/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!str_detect(protein_names, "Trypsin|Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin|Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger|salivary|Zinc|Prolactin|Junction|Gliomedin|Desmo|Cystein|Polyadenylate")) %>%
  filter(!is.na(protein_names)) %>%
  filter(!experiment == "P0442_77min_Blank") %>%
  dplyr::select(length, experiment, intensity) %>%
  mutate(experiment = gsub('P0441_77min', 'Original_bone', experiment))
bleach <- read_delim("khudji_bleach/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(experiment == "Blank") %>%
  dplyr::select(length, experiment, intensity) %>%
  mutate(experiment = gsub('Blank', 'Bleach', experiment))
bleach_bone <- read_delim("khudji_bleach/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
filter(!str_detect(protein_names, "Trypsin|Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin|Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger|salivary|Zinc|Prolactin|Junction|Gliomedin|Desmo|Cystein|Collectin|Leucine|Baculoviral")) %>%
  filter(experiment == "Blank") %>%
  dplyr::select(length, experiment, intensity) %>%
  mutate(experiment = gsub('Blank', 'Bleach_bone', experiment))

# Combine all
length <- full_join(original, 
                      full_join(original_bone, 
                                full_join(bleach, bleach_bone))) %>%
  filter( intensity > 0)

# Sum intensity by length
length_summary <- length %>%
  group_by(experiment, length) %>%
  dplyr::summarize(total_intensity = sum(intensity))

# Plot barcharts (creates Figure S3A)
ggplot(length_summary, aes(x=length, y=total_intensity, fill=experiment)) + 
  geom_bar(stat="identity") +
  facet_wrap(~experiment, ncol=1) +
  scale_fill_manual(values=met.brewer("Cassatt2", 4), name="Proteome") +
  xlab("Peptide length") +
  ylab("Total intensity") +
  theme_bw() +  
  theme(strip.background = element_blank(), strip.text = element_blank())

# Extract comparisons
full <- length %>%
  filter(experiment %in% c("Original", "Bleach"))
bone <- length %>%
  filter(experiment %in% c("Original_bone", "Bleach_bone"))

# Find Box-Cox transformation of the data
MASS::boxcox(lm(length ~ experiment, data=bone))

# Checking transformation
ggqqplot(bone$length)
hist(bone$length)

# Additive model 
model1 <- lm(length ~ experiment, weights=intensity, data = bone)

# Null model
model0 <- lm(length ~ 1, weights=intensity, data = bone)

# Compare models
anova(model1, model0) 

# Check model
plot(model1)
qqnorm(resid(model1))
qqline(resid(model1)) 
```

## Deamidation

The deamidation values were calculated from the evidence.txt file using the Python script published in:

Mackie, M., Rüther, P., Samodova, D., Di Gianvincenzo, F., Granzotto, C., Lyon, D., Peggie, D. A., Howard, H., Harrison, L., Jensen, L. J., Olsen, J. V., & Cappellini, E. (2018). Palaeoproteomic Profiling of Conservation Layers on a 14th Century Italian Wall Painting. Angewandte Chemie, 57(25), 7369–7374.

The output from the calculation is in the data-folder in the GitHub/Zenodo repository, so you can just run the following code after downloading the entire repository following the instructions at the start.

```{r}
# Load output from the Mackie et al. script
deamidation <- read.csv("data/deamidation_khudji.txt", sep="") %>%
  clean_names()

# Edit sample names
deamidation <- deamidation %>%
   mutate(raw_file = gsub('20230614_EXPL8_nLC11_GT_collab_77min_DDA_P0441', 'original', raw_file)) %>%
   mutate(raw_file = gsub('20231005_EXPL5_nLC11_GT_collab_77min_DDA_Blank_P0581', 'bleach', raw_file))

# Fix version column
deamidation <- deamidation %>%
   mutate(version = gsub('original_bone', 'bone', version)) %>%
   mutate(version = gsub('bleach_bone', 'bone', version)) %>%
   mutate(version = gsub('bleach', 'full', version)) %>%
   mutate(version = gsub('original', 'full', version))

# Plot (creates Figure 6D)
ggplot(deamidation, aes(x=raw_file, y=mean, fill=version)) + 
  geom_bar(stat="identity", position="dodge") +
  geom_errorbar(aes(ymin=mean-std, ymax=mean+std), width=.2,
                 position=position_dodge(.9)) +
  scale_fill_manual(values=met.brewer("Cassatt2", 2), name="Proteome") +
  facet_grid(~n_q) +
  xlab("") + 
  ylab("Deamidation") +
  theme_bw() 
```

## Hydrophobicity

```{r}
# Load data
original <- read_delim("khudji_original/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!experiment == "P0442_77min_Blank") %>%
  dplyr::select(sequence, experiment, intensity) %>%
  mutate(experiment = gsub('P0441_77min', 'Original', experiment))
original_bone <- read_delim("khudji_original/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!str_detect(protein_names, "Trypsin|Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin|Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger|salivary|Zinc|Prolactin|Junction|Gliomedin|Desmo|Cystein|Polyadenylate")) %>%
  filter(!is.na(protein_names)) %>%
  filter(!experiment == "P0442_77min_Blank") %>%
  dplyr::select(sequence, experiment, intensity) %>%
  mutate(experiment = gsub('P0441_77min', 'Original_bone', experiment))
bleach <- read_delim("khudji_bleach/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(experiment == "Blank") %>%
  dplyr::select(sequence, experiment, intensity) %>%
  mutate(experiment = gsub('Blank', 'Bleach', experiment))
bleach_bone <- read_delim("khudji_bleach/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
filter(!str_detect(protein_names, "Trypsin|Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin|Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger|salivary|Zinc|Prolactin|Junction|Gliomedin|Desmo|Cystein|Collectin|Leucine|Baculoviral")) %>%
  filter(experiment == "Blank") %>%
  dplyr::select(sequence, experiment, intensity) %>%
  mutate(experiment = gsub('Blank', 'Bleach_bone', experiment))

# Combine files
evidence <- full_join(original, 
                      full_join(original_bone, 
                                full_join(bleach, bleach_bone))) %>%
  filter( intensity > 0)

# Calculate hydrophobicity
evidence <- evidence %>% 
  filter(intensity > 0) %>%
  mutate(hydro = hydrophobicity(sequence, scale = "KyteDoolittle"))

# Summarize
hydro_summary <- evidence %>%
  group_by(experiment) %>%
  dplyr::summarise(mean_hydro = weighted.mean(x=hydro, w=intensity, na.rm=TRUE),
                   weighted_var= wtd.var(x=hydro, w=intensity, na.rm=TRUE))
hydro_summary$weighted_sd <- sqrt(hydro_summary$weighted_var)

# Edit labels
hydro_summary <- hydro_summary %>%
  separate(experiment, c("experiment", "proteome"), sep="_") %>%
  mutate(proteome = replace_na(proteome, "full"))

# Plot (creates Figure S3B)
ggplot(hydro_summary, aes(x=experiment, y=mean_hydro, colour=proteome)) + 
  geom_point(position=position_dodge(.9), size=5) +
  geom_errorbar(aes(ymin=mean_hydro-weighted_sd, ymax=mean_hydro+weighted_sd), width=.2,
                 position=position_dodge(.9)) +
  scale_colour_manual(values=met.brewer("Cassatt2", 2), name="Proteome") +
  xlab("") +
  ylab("Hydrophobicity") +
  theme_bw()

# Extract comparisons
full <- evidence %>%
  filter(experiment %in% c("Original", "Bleach"))
bone <- evidence %>%
  filter(experiment %in% c("Original_bone", "Bleach_bone"))

# Find Box-Cox transformation of the data
MASS::boxcox(lm(hydro+3 ~ experiment, data=bone))

# Checking transformation
ggqqplot(bone$hydro)
hist(bone$hydro)

# Additive model 
model1 <- lm(hydro ~ experiment, weights=intensity, data = bone)

# Null model
model0 <- lm(hydro ~ 1, weights=intensity, data = bone)

# Compare models
anova(model1, model0) 

# Check model
plot(model1)
qqnorm(resid(model1))
qqline(resid(model1)) 
```

## Proteome reconstruction

This part is based on output files from PEAKS11, which can be found on the ProteomeXchange Consortium via the PRIDE partner repository with the dataset identifier PXD050370. To run the code, you need to download the files from PRIDE, and move them to the same folder as where this script is located. 

The reconstruction is done twice, once for each extraction method, which is then checked and a consensus sequence created manually (see publication for details).

### Bleach

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Load protein-level data
spider_proteins <- read_delim("bleach_fdr1/spider.proteins.csv", 
    delim = ",", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() 

# Filter for 5 or more peptides
spider_proteins <- spider_proteins %>%
  filter(number_peptides > 4)

# Remove contaminants
spider_proteins <- spider_proteins %>%
  filter(!str_detect(accession, "CONTAM")) %>%
  filter(!str_detect(description, "Keratin")) %>%
  filter(!str_detect(accession, "UBB_HUMAN|FILA_HUMAN|DSG1_HUMAN|RL40_HUMAN|CASPE_HUMAN|UBC_HUMAN|CALL5_HUMAN|DESP_HUMAN|DCD_HUMAN|HORN_HUMAN|CYTA_HUMAN|RS27A_HUMAN|ALBU_HUMAN|G3P_HUMAN|A0A2R8Y422_HUMAN|K2C1_HUMAN|K2C5_HUMAN|K1C14_HUMAN|K2C6B_HUMAN|K1C10_HUMAN|K1C9_HUMAN|K2C75_HUMAN|K22E_HUMAN|K1H1_HUMAN|K1C16_HUMAN|K1C15_HUMAN|K2C6C_HUMAN|HORN_Nean|K2C6A_HUMAN|K1C17_HUMAN|K2C1B_HUMAN|EMID1_HUMAN|Q5KU26|COL12_HUMAN|KT33A_HUMAN"))
```

Now we go through the extracts one by one, and reconstruct the sequences separately. 

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Load peptide-level data
peptide <- read_delim("bleach_fdr1/spider.protein-peptides.csv", delim = ",", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() 

# Create vector of proteins
protein_recon <- spider_proteins$accession

# Create blank data frame new seq's will go into
new_fasta <- data.frame(matrix(ncol = 2, nrow = 0))
colnames(new_fasta) <- c('protein', 'sequence')


# For loop to pull reconstruct proteins and add to file
for (protein in protein_recon) { 

# Select only wanted protein
target <- peptide %>%
  filter(accession == protein) %>%
  dplyr::select(peptide, start, end) 

# Clean up peptides by removing parentheses
target$peptide <- gsub("\\s*\\([^\\)]+\\)","", target$peptide)

# Expand numbers between start and end to get a dataframe of all covered positions
target <- target %>%
  mutate(covered = map2(start, end, seq, by=1))  
pos_list <- target %>% dplyr::select(covered) %>%
  mutate(covered = map(covered, ~as_data_frame(t(.x)))) %>% unnest() %>%
  gather(remove, position, contains("V")) %>%
  dplyr::select(-remove) %>%
  mutate(ID =row_number())

# Make similar dataframe of all peptides
peptide_sep <- target %>% dplyr::select(peptide) 
peptide_sep <- data.frame(str_split_fixed(peptide_sep$peptide, "", max(nchar(peptide_sep$peptide)))) %>%
  gather(remove, aa, contains("X")) %>%
  dplyr::select(-remove) %>%
  mutate(ID = row_number())

# Combine position and aa dataframes
pos_aa <- full_join(pos_list, peptide_sep) %>%
  dplyr::select(-ID) %>%
  filter(!is.na(position))

# Collapse amino acids
pos_aa <- pos_aa %>%
  group_by(position) %>%
  dplyr::summarise(amino_acids = paste(aa, collapse = ""))

# Call an amino acid for each position
pos_aa <- pos_aa %>%
  rowwise %>%
  mutate(count = length(unique(strsplit(amino_acids, "")[[1]])))
## Function for finding most common amino acid 
most_repeated_character <- function(x) { 
  tab <- table(strsplit(x, "")[[1]])
  names(tab)[tab == max(tab)]
}
pos_aa <- pos_aa %>%
  mutate(consensus = ifelse(count == 1,
                          unique(strsplit(amino_acids, "")[[1]]),
                          most_repeated_character(amino_acids))) %>%
  dplyr::select(-count, -amino_acids)

# This will make all proteins 1500 aa's long (or wherever the last peptide is), which then needs to be fixed manually, as the author seems incapable of figuring out how to pull this detail from the human reference proteome computationally. Sorry.
max_pos <- 1500

# Expand to fill in empty positions
pos_aa <- pos_aa %>% 
  type.convert(as.is=TRUE) %>% 
  complete(position = 1:max_pos) %>%
  mutate(consensus = replace_na(consensus, "-"))

# Make consensus sequence
consensus <- pos_aa %>%
  pivot_wider(names_from = "position", values_from = "consensus") %>%
  unite(sequence, is.character, sep="")
consensus$protein <- protein

## Add to dataframe
new_fasta <- rbind(consensus, new_fasta)
}

# Filter for unique proteins
new_fasta <- new_fasta %>%
  distinct(sequence, .keep_all=TRUE)

# Make the dataframe into a fasta
writeFasta<-function(data, filename){
  fastaLines = c()
  for (rowNum in 1:nrow(data)){
    fastaLines = c(fastaLines, as.character(paste(">", data[rowNum,"protein"], sep = "")))
    fastaLines = c(fastaLines,as.character(data[rowNum,"sequence"]))
  }
  fileConn<-file(filename)
  writeLines(fastaLines, fileConn)
  close(fileConn)
}
writeFasta(new_fasta, "output_bleach.fasta")
```

### Original

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Load protein-level data
spider_proteins <- read_delim("original_fdr1/spider.proteins.csv", 
    delim = ",", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() 

# Filter for 5 or more peptides
spider_proteins <- spider_proteins %>%
  filter(number_peptides > 4)

# Remove contaminants
spider_proteins <- spider_proteins %>%
  filter(!str_detect(accession, "CONTAM")) %>%
  filter(!str_detect(description, "Keratin")) %>%
  filter(!str_detect(accession, "UBB_HUMAN|FILA_HUMAN|DSG1_HUMAN|RL40_HUMAN|CASPE_HUMAN|UBC_HUMAN|CALL5_HUMAN|DESP_HUMAN|DCD_HUMAN|HORN_HUMAN|CYTA_HUMAN|RS27A_HUMAN|ALBU_HUMAN|G3P_HUMAN|A0A2R8Y422_HUMAN|K2C1_HUMAN|K2C5_HUMAN|K1C14_HUMAN|K2C6B_HUMAN|K1C10_HUMAN|K1C9_HUMAN|K2C75_HUMAN|K22E_HUMAN|K1H1_HUMAN|K1C16_HUMAN|K1C15_HUMAN|K2C6C_HUMAN|HORN_Nean|K2C6A_HUMAN|K1C17_HUMAN|K2C1B_HUMAN|EMID1_HUMAN|Q5KU26|COL12_HUMAN|KT33A_HUMAN"))
```

Now we go through the extracts one by one, and reconstruct the sequences separately. 

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Load peptide-level data
peptide <- read_delim("original_fdr1/spider.protein-peptides.csv", delim = ",", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() 

# Create vector of proteins
protein_recon <- spider_proteins$accession

# Create blank data frame new seq's will go into
new_fasta <- data.frame(matrix(ncol = 2, nrow = 0))
colnames(new_fasta) <- c('protein', 'sequence')


# For loop to pull reconstruct proteins and add to file
for (protein in protein_recon) { 

# Select only wanted protein
target <- peptide %>%
  filter(accession == protein) %>%
  dplyr::select(peptide, start, end) 

# Clean up peptides by removing parentheses
target$peptide <- gsub("\\s*\\([^\\)]+\\)","", target$peptide)

# Expand numbers between start and end to get a dataframe of all covered positions
target <- target %>%
  mutate(covered = map2(start, end, seq, by=1))  
pos_list <- target %>% dplyr::select(covered) %>%
  mutate(covered = map(covered, ~as_data_frame(t(.x)))) %>% unnest() %>%
  gather(remove, position, contains("V")) %>%
  dplyr::select(-remove) %>%
  mutate(ID =row_number())

# Make similar dataframe of all peptides
peptide_sep <- target %>% dplyr::select(peptide) 
peptide_sep <- data.frame(str_split_fixed(peptide_sep$peptide, "", max(nchar(peptide_sep$peptide)))) %>%
  gather(remove, aa, contains("X")) %>%
  dplyr::select(-remove) %>%
  mutate(ID = row_number())

# Combine position and aa dataframes
pos_aa <- full_join(pos_list, peptide_sep) %>%
  dplyr::select(-ID) %>%
  filter(!is.na(position))

# Collapse amino acids
pos_aa <- pos_aa %>%
  group_by(position) %>%
  dplyr::summarise(amino_acids = paste(aa, collapse = ""))

# Call an amino acid for each position
pos_aa <- pos_aa %>%
  rowwise %>%
  mutate(count = length(unique(strsplit(amino_acids, "")[[1]])))
## Function for finding most common amino acid 
most_repeated_character <- function(x) { 
  tab <- table(strsplit(x, "")[[1]])
  names(tab)[tab == max(tab)]
}
pos_aa <- pos_aa %>%
  mutate(consensus = ifelse(count == 1,
                          unique(strsplit(amino_acids, "")[[1]]),
                          most_repeated_character(amino_acids))) %>%
  dplyr::select(-count, -amino_acids)

# This will make all proteins 1500 aa's long (or wherever the last peptide is), which then needs to be fixed manually, as the author seems incapable of figuring out how to pull this detail from the human reference proteome computationally. Sorry.
max_pos <- 1500

# Expand to fill in empty positions
pos_aa <- pos_aa %>% 
  type.convert(as.is=TRUE) %>% 
  complete(position = 1:max_pos) %>%
  mutate(consensus = replace_na(consensus, "-"))

# Make consensus sequence
consensus <- pos_aa %>%
  pivot_wider(names_from = "position", values_from = "consensus") %>%
  unite(sequence, is.character, sep="")
consensus$protein <- protein

## Add to dataframe
new_fasta <- rbind(consensus, new_fasta)
}

# Filter for unique proteins
new_fasta <- new_fasta %>%
  distinct(sequence, .keep_all=TRUE)

# Make the dataframe into a fasta
writeFasta<-function(data, filename){
  fastaLines = c()
  for (rowNum in 1:nrow(data)){
    fastaLines = c(fastaLines, as.character(paste(">", data[rowNum,"protein"], sep = "")))
    fastaLines = c(fastaLines,as.character(data[rowNum,"sequence"]))
  }
  fileConn<-file(filename)
  writeLines(fastaLines, fileConn)
  close(fileConn)
}
writeFasta(new_fasta, "output_original.fasta")
```
