---
title: "Decontamination of the Khudji tooth"
author: "Zandra Fagernäs"
---

This notebook contains analyses for decontamination and proteome reconstruction of the deciduous hominin tooth from the manuscript "Cleaning the Dead: Optimized decontamination enhances palaeoproteomic analyses of a Pleistocene hominin tooth from Khudji, Tajikistan" by Fagernäs et al.

```{r echo=FALSE, message=FALSE, warning=FALSE}
library(MetBrewer)
library(janitor)
library(ggpubr)
library(Hmisc)
library(ape)
library(Peptides)
library(tidyverse)
```

## Number of proteins and peptides recovered

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.width=4, fig.height=4}
# Load text file with protein groups and clean 
original <- read_delim("<PATH_TO>/proteinGroups.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!str_detect(protein_names, "Trypsin")) %>%
  filter(!str_detect(protein_i_ds, "CON__P00761")) %>%
  dplyr::select(protein_i_ds, protein_names, razor_unique_peptides_p0441_77min) %>%
  filter(razor_unique_peptides_p0441_77min > 1)
bleach <- read_delim("<PATH_TO>/proteinGroups.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+"))  %>%
  filter(!str_detect(protein_names, "Trypsin")) %>%
  filter(!str_detect(protein_i_ds, "CON__P00761")) %>%
  dplyr::select(protein_i_ds, protein_names, razor_unique_peptides_blank) %>%
  filter(razor_unique_peptides_blank > 1)

# Combine files
protein <- full_join(original, bleach)

# Assign origin of protein
protein <- protein %>%
  mutate(skin = str_detect(protein_names, 
       "Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin")) %>%
  mutate(bone = str_detect(protein_names, "Collagen|EMI")) %>%
  mutate(other = str_detect(protein_names, "Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger")) %>%
  mutate(tissue = if_else(skin == TRUE, "skin", 
                          if_else(bone == TRUE, "bone", "other"))) %>%
  select(-skin, -bone, -other)

# Long form
protein <- protein %>%
  pivot_longer(names_to = "sample", values_to = "peptides", 3:4) %>%
  filter(peptides > 0)

# Clean up sample names
protein <- protein %>%
    mutate(sample = gsub('razor_unique_peptides_p0441_77min', 'Original', sample)) %>%
    mutate(sample = gsub('razor_unique_peptides_blank', 'Bleach', sample))

# Count proteins 
protein_summary <- protein %>%
  group_by(sample, tissue) %>%
  summarize(proteins = sum(peptides > 0),
            peptides = sum(peptides))

# Long format
protein_summary <- protein_summary %>%
  pivot_longer(names_to = "measure", values_to = "count", 3:4)

# Plot 
summary_fig <- ggplot(protein_summary, aes(x=sample, y=count, fill=tissue)) +
  geom_bar(position="fill", stat = "identity") +
  facet_wrap(~measure, scales = "free_y") +
  scale_fill_manual(values=met.brewer("Cassatt2", 3)) +
  theme_bw()
```

## Peptide length 

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Load data
original <- read_delim("<PATH_TO>/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!experiment == "P0442_77min_Blank") %>%
  dplyr::select(length, experiment, intensity) %>%
  mutate(experiment = gsub('P0441_77min', 'Original', experiment))
original_bone <- read_delim("<PATH_TO>/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!str_detect(protein_names, "Trypsin|Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin|Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger|salivary|Zinc|Prolactin|Junction|Gliomedin|Desmo|Cystein|Polyadenylate")) %>%
  filter(!is.na(protein_names)) %>%
  filter(!experiment == "P0442_77min_Blank") %>%
  dplyr::select(length, experiment, intensity) %>%
  mutate(experiment = gsub('P0441_77min', 'Original_bone', experiment))
bleach <- read_delim("<PATH_TO>/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(experiment == "Blank") %>%
  dplyr::select(length, experiment, intensity) %>%
  mutate(experiment = gsub('Blank', 'Bleach', experiment))
bleach_bone <- read_delim("<PATH_TO>/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
filter(!str_detect(protein_names, "Trypsin|Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin|Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger|salivary|Zinc|Prolactin|Junction|Gliomedin|Desmo|Cystein|Collectin|Leucine|Baculoviral")) %>%
  filter(experiment == "Blank") %>%
  dplyr::select(length, experiment, intensity) %>%
  mutate(experiment = gsub('Blank', 'Bleach_bone', experiment))

# Combine all
length <- full_join(original, 
                      full_join(original_bone, 
                                full_join(bleach, bleach_bone))) %>%
  filter( intensity > 0)

# Sum intensity by length
length <- length %>%
  group_by(experiment, length) %>%
  dplyr::summarize(total_intensity = sum(intensity))

# Plot barcharts
length_plot <- ggplot(length, aes(x=length, y=total_intensity, fill=experiment)) + 
  geom_bar(stat="identity") +
  facet_wrap(~experiment, ncol=1) +
  scale_fill_manual(values=met.brewer("Cassatt2", 4), name="Proteome") +
  xlab("Peptide length") +
  ylab("Total intensity") +
  theme_bw() +  
  theme(strip.background = element_blank(), strip.text = element_blank())
```

## Deamidation

Based on python script from Mackie et al. 2018.

```{r}
# Load data
deamidation <- read.csv("<PATH_TO>/deamidation.txt", sep="") %>%
  clean_names()

# Edit sample names
deamidation <- deamidation %>%
   mutate(raw_file = gsub('20230614_EXPL8_nLC11_GT_collab_77min_DDA_P0441', 'original', raw_file)) %>%
   mutate(raw_file = gsub('20231005_EXPL5_nLC11_GT_collab_77min_DDA_Blank_P0581', 'bleach', raw_file))

# Fix version column
deamidation <- deamidation %>%
   mutate(version = gsub('original_bone', 'bone', version)) %>%
   mutate(version = gsub('bleach_bone', 'bone', version)) %>%
   mutate(version = gsub('bleach', 'full', version)) %>%
   mutate(version = gsub('original', 'full', version))

# Plot
deamidation_fig <- ggplot(deamidation, aes(x=raw_file, y=mean, fill=version)) + 
  geom_bar(stat="identity", position="dodge") +
  geom_errorbar(aes(ymin=mean-std, ymax=mean+std), width=.2,
                 position=position_dodge(.9)) +
  scale_fill_manual(values=met.brewer("Cassatt2", 2), name="Proteome") +
  facet_grid(~n_q) +
  xlab("") + 
  ylab("Deamidation") +
  theme_bw() 
```

## Hydrophobicity

```{r}
# Load data
original <- read_delim("<PATH_TO>/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!experiment == "P0442_77min_Blank") %>%
  dplyr::select(sequence, experiment, intensity) %>%
  mutate(experiment = gsub('P0441_77min', 'Original', experiment))
original_bone <- read_delim("<PATH_TO>/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(!str_detect(protein_names, "Trypsin|Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin|Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger|salivary|Zinc|Prolactin|Junction|Gliomedin|Desmo|Cystein|Polyadenylate")) %>%
  filter(!is.na(protein_names)) %>%
  filter(!experiment == "P0442_77min_Blank") %>%
  dplyr::select(sequence, experiment, intensity) %>%
  mutate(experiment = gsub('P0441_77min', 'Original_bone', experiment))
bleach <- read_delim("<PATH_TO>/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
  filter(experiment == "Blank") %>%
  dplyr::select(sequence, experiment, intensity) %>%
  mutate(experiment = gsub('Blank', 'Bleach', experiment))
bleach_bone <- read_delim("<PATH_TO>/evidence.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() %>%
  filter(!reverse %in% c("+")) %>%
filter(!str_detect(protein_names, "Trypsin|Keratin|Serum|Filaggrin|Hornerin|Cystatin-A|Dermcidin|Desmoglein-1|Calmodulin|Serpin|Caspase|Suprabasin|Alpha-amylase|Ig |Ubiquitin|Lysozyme| C1q|Scavenger|salivary|Zinc|Prolactin|Junction|Gliomedin|Desmo|Cystein|Collectin|Leucine|Baculoviral")) %>%
  filter(experiment == "Blank") %>%
  dplyr::select(sequence, experiment, intensity) %>%
  mutate(experiment = gsub('Blank', 'Bleach_bone', experiment))

# Combine files
evidence <- full_join(original, 
                      full_join(original_bone, 
                                full_join(bleach, bleach_bone))) %>%
  filter( intensity > 0)

# Calculate hydrophobicity
evidence <- evidence %>% 
  filter(intensity > 0) %>%
  mutate(hydro = hydrophobicity(sequence, scale = "KyteDoolittle"))

# Summarize
hydro_summary <- evidence %>%
  group_by(experiment) %>%
  dplyr::summarise(mean_hydro = weighted.mean(x=hydro, w=intensity, na.rm=TRUE),
                   weighted_var= wtd.var(x=hydro, w=intensity, na.rm=TRUE))
hydro_summary$weighted_sd <- sqrt(hydro_summary$weighted_var)

# Edit labels
hydro_summary <- hydro_summary %>%
  separate(experiment, c("experiment", "proteome"), sep="_") %>%
  mutate(proteome = replace_na(proteome, "full"))

# Plot
hydro_fig <- ggplot(hydro_summary, aes(x=experiment, y=mean_hydro, colour=proteome)) + 
  geom_point(position=position_dodge(.9), size=5) +
  geom_errorbar(aes(ymin=mean_hydro-weighted_sd, ymax=mean_hydro+weighted_sd), width=.2,
                 position=position_dodge(.9)) +
  scale_colour_manual(values=met.brewer("Cassatt2", 2), name="Proteome") +
  xlab("") +
  ylab("Hydrophobicity") +
  theme_bw()
```

## Proteome reconstruction

Based on output from PEAKS. First, we get a list of proteins with more than 5 peptides.

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Load protein-level data
protein_original <- read_delim("<PATH_TO>/proteins.csv", 
    delim = ",", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() 
protein_bleach <- read_delim("<PATH_TO>/proteins.csv", 
    delim = ",", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() 

# Filter for >5 peptides
protein_original <- protein_original %>%
  filter(number_peptides > 5) %>%
  dplyr::select(accession, number_peptides, coverage_percent)
protein_bleach <- protein_bleach %>%
  filter(number_peptides > 5) %>%
  dplyr::select(accession, number_peptides, coverage_percent)

# Remove contaminants (there were none in the bleached one)
protein_original <- protein_original %>%
  filter(accession %in% c("sp|P02452|CO1A1_HUMAN", "COL1A1_variant", "COL1A2_Nean", "COL2A1_Xiahe1", "sp|P05997|CO5A2_HUMAN", "sp|P02461|CO3A1_HUMAN", "COL3A1_Denisova", "COL3A1_Nean", "sp|P29400|CO4A5_HUMAN", "COL4A1_Nean", "sp|Q03692|COAA1_HUMAN", "COL24A1_Nean", "sp|Q17RW2|COOA1_HUMAN"))
```

Now we go through the extracts one by one, and reconstruct the sequences separately. Later, I align them and reconstruct the full sequence in Geneious.

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Load peptide-level data
peptide <- read_delim("<PATH_TO>/protein-peptides.csv", delim = ",", escape_double = FALSE, trim_ws = TRUE) %>%
  clean_names() 

# Create vector of proteins
protein_recon <- protein_combined$accession

# Create blank data frame new seq's will go into
new_fasta <- data.frame(matrix(ncol = 2, nrow = 0))
colnames(new_fasta) <- c('protein', 'sequence')

# Load human proteome
human <- readAAStringSet("<PATH_TO>/uniprot_human_proteome_20220117.fasta", format="fasta")

# Make dataframe of human protein length
length_df <- data.frame(names(human), width(human))
colnames(length_df) <- c("protein", "width")


# For loop to pull reconstruct proteins and add to file
for (protein in protein_recon) { 

# Select only wanted protein
target <- peptide %>%
  filter(protein_accession == protein) %>%
  dplyr::select(peptide, start, end) 

# Clean up peptides by removing parentheses
target$peptide <- gsub("\\s*\\([^\\)]+\\)","", target$peptide)

# Clean up peptides by removing amino acids outside dots
target <- target %>%
  separate(peptide, c("remove1", "peptide", "remove2"), sep="\\.") %>%
  dplyr::select(-remove1, -remove2)

# Expand numbers between start and end to get a dataframe of all covered positions
target <- target %>%
  mutate(covered = map2(start, end, seq, by=1))  
pos_list <- target %>% dplyr::select(covered) %>%
  mutate(covered = map(covered, ~as_data_frame(t(.x)))) %>% unnest() %>%
  gather(remove, position, contains("V")) %>%
  dplyr::select(-remove) %>%
  mutate(ID =row_number())

# Make similar dataframe of all peptides
peptide_sep <- target %>% dplyr::select(peptide) 
peptide_sep <- data.frame(str_split_fixed(peptide_sep$peptide, "", max(nchar(peptide_sep$peptide)))) %>%
  gather(remove, aa, contains("X")) %>%
  dplyr::select(-remove) %>%
  mutate(ID = row_number())

# Combine position and aa dataframes
pos_aa <- full_join(pos_list, peptide_sep) %>%
  dplyr::select(-ID) %>%
  filter(!is.na(position))

# Collapse amino acids
pos_aa <- pos_aa %>%
  group_by(position) %>%
  summarise(amino_acids = paste(aa, collapse = ""))

# Call an amino acid for each position
pos_aa <- pos_aa %>%
  rowwise %>%
  mutate(count = length(unique(strsplit(amino_acids, "")[[1]])))
## Function for finding most common amino acid 
most_repeated_character <- function(x) { 
  tab <- table(strsplit(x, "")[[1]])
  names(tab)[tab == max(tab)]
}
pos_aa <- pos_aa %>%
  mutate(consensus = ifelse(count == 1,
                          unique(strsplit(amino_acids, "")[[1]]),
                          most_repeated_character(amino_acids))) %>%
  dplyr::select(-count, -amino_acids)

# Find out how long the protein is in modern humans - doesn't work, adding to 1500 for all and manually fixing! Sorry to anyone using this script, I didn't have time to fix this. Feel free to email me and complain.
max_pos <- 1500

# Expand to fill in empty positions
pos_aa <- pos_aa %>% 
  type.convert(as.is=TRUE) %>% 
  complete(position = 1:max_pos) %>%
  mutate(consensus = replace_na(consensus, "-"))

# Make consensus sequence
consensus <- pos_aa %>%
  pivot_wider(names_from = "position", values_from = "consensus") %>%
  unite(sequence, is.character, sep="")
consensus$protein <- protein

## Add to dataframe
new_fasta <- rbind(consensus, new_fasta)
}

# Filter for unique proteins
new_fasta <- new_fasta %>%
  distinct(sequence, .keep_all=TRUE)

# Make the dataframe into a fasta
writeFasta<-function(data, filename){
  fastaLines = c()
  for (rowNum in 1:nrow(data)){
    fastaLines = c(fastaLines, as.character(paste(">", data[rowNum,"protein"], sep = "")))
    fastaLines = c(fastaLines,as.character(data[rowNum,"sequence"]))
  }
  fileConn<-file(filename)
  writeLines(fastaLines, fileConn)
  close(fileConn)
}

writeFasta(new_fasta, "<PATH_TO>/khudji.fasta")
```